src/config.cpp

```diff
@@ -16,120 +16,231 @@ ONNXTensorElementDataType TranslateTensorType(std::string_view value) {
+  Decoder_Element decoder_{v_.decoder};
 };
 
-struct Root_Element : JSON::Element {
-  explicit Root_Element(Config& config) : config_{config} {}
+struct Search_Element : JSON::Element {
+  explicit Search_Element(Config::Search& v) : v_{v} {}
 
   void OnString(std::string_view name, std::string_view value) override {
-    if (name == "tokenizer_class") {
-      config_.tokenizer_class = value;
-    } else if (name == "prefix") {
-      config_.prefix = value;
-    }
+    throw JSON::unknown_value_error{};
   }
 
   void OnNumber(std::string_view name, double value) override {
-    // Sequence Generation
-    if (name == "max_length") {
-      config_.max_length = static_cast<int>(value);
-    } else if (name == "min_length") {
-      config_.min_length = static_cast<int>(value);
+    if (name == "min_length") {
+      v_.min_length = static_cast<int>(value);
+    } else if (name == "max_length") {
+      v_.max_length = static_cast<int>(value);
     } else if (name == "num_beams") {
-      config_.num_beams = static_cast<int>(value);
-    } else if (name == "temperature") {
-      config_.temperature = static_cast<float>(value);
+      v_.num_beams = static_cast<int>(value);
+    } else if (name == "num_return_sequences") {
+      v_.num_return_sequences = static_cast<int>(value);
     } else if (name == "top_k") {
-      config_.top_k = static_cast<int>(value);
+      v_.top_k = static_cast<int>(value);
     } else if (name == "top_p") {
-      config_.top_p = static_cast<float>(value);
+      v_.top_p = static_cast<float>(value);
+    } else if (name == "temperature") {
+      v_.temperature = static_cast<float>(value);
     } else if (name == "repetition_penalty") {
-      config_.repetition_penalty = static_cast<float>(value);
+      v_.repetition_penalty = static_cast<float>(value);
     } else if (name == "length_penalty") {
-      config_.length_penalty = static_cast<float>(value);
-
-      // Tokenizer Parameters
-    } else if (name == "pad_token_id") {
-      config_.pad_token_id = static_cast<int>(value);
-    } else if (name == "eos_token_id") {
-      config_.eos_token_id = static_cast<int>(value);
-    } else if (name == "bos_token_id") {
-      config_.bos_token_id = static_cast<int>(value);
-    } else if (name == "decoder_start_token_id") {
-      config_.decoder_start_token_id = static_cast<int>(value);
-    } else if (name == "sep_token_id") {
-      config_.sep_token_id = static_cast<int>(value);
-    }
+      v_.length_penalty = static_cast<float>(value);
+    } else if (name == "no_repeat_ngram_size") {
+      v_.no_repeat_ngram_size = static_cast<int>(value);
+    } else if (name == "diversity_penalty") {
+      v_.diversity_penalty = static_cast<float>(value);
+    } else if (name == "length_penalty") {
+      v_.length_penalty = static_cast<float>(value);
+    } else
+      throw JSON::unknown_value_error{};
   }
 
   void OnBool(std::string_view name, bool value) override {
     if (name == "early_stopping") {
```

Comment:
Is this needed if `early_stopping = true` by default?

Reply:
If you want to set it to false, perhaps?